lib/plugin/routes/public/teams.js

```diff
@@ -406,7 +406,11 @@ exports.register = function (server, options, next) {
       const limit = request.query.limit || server.udaruConfig.get('authorization.defaultPageSize')
       const page = request.query.page || 1
 
-      request.udaruCore.teams.listUsers({ id, page, limit, organizationId }, reply)
+      request.udaruCore.teams.read({ id, organizationId }, (err) => {
```

Comment:
If we're going on this route it makes sense that every endpoint that has a team id in it to verify first if that team exists. Probably some of the internal functions have this incorporated but for all this other this should be called. And I'm pretty sure it's not only one...
Some smarter approach would be to have this automatically checked (to separate it from the handler) as part of the plugin. You just need to point which function to use for check and where to pick the id/org id.